### PR TITLE
4864 back date UI

### LINKED ISSUE #4864: Backdating prescription UI
## Is your feature request related to a problem? Please describe 👀
A tweak to the UI please.
The date field is floating in the middle of the toolbar area, not vertically aligned with the other inputs:

image

Can we adjust so that the inputs align, and bring the date input closer to the others

image

## Describe the solution you'd like 🎁
In addition to the UI move above, I'd also like the following changes:

I'd also like to default the value to today's date - leaving the input empty invites the user to enter something, which would be the wrong intention. Defaulting to today gives the user a clue that the date is the date of the prescription and therefore will be today's date by default.

Additional thoughts
* maybe move it to the 'more info' panel?
* if so, shift clinician up to make the toolbar smaller vertically
* rename? The `Date` label isn't telling me what the intention is. Perhaps if we move to the info panel we could add some help text?

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #4864

# 👩🏻‍💻 What does this PR do?

Aligns datepicker with rest of the inputs in dispensing.

image

Defaults `prescription date` to created datetime

Sets picked_date to prescription date if we change the date time for a prescription (by setting it at verified time)

Disables date input after verified

## 💌 Any notes for the reviewer?

Do we still want the prescription date in the more tab?

# 🧪 Testing

- [ ] Start dispensing, check date picker shows up correctly..

# 📃 Documentation

- [X] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour

### CODE REVIEW

**Comment by lache-melvin** [comment]
This is to go into 2.3 right?

**Comment by jmbrunskill** [comment]
> This is to go into 2.3 right?

Updated...

File: client/packages/invoices/src/Prescriptions/DetailView/Toolbar.tsx [hunk 1]
```diff
@@ -69,11 +75,17 @@ export const Toolbar: FC = () => {
               }
             />
           </Box>
-          <Box display="flex" flex={1}>
+          <Box
+            display="flex"
+            flexDirection="column"
+            flex={1}
+            marginLeft={'8px'}
+          >
```
Comment (mark-prins) [inline client/packages/invoices/src/Prescriptions/DetailView/Toolbar.tsx:83]:
```suggestion
          <Box display="flex" flexDirection="column" flex={1} marginLeft={3}>
```

two things - we need to remove hard coded pixel values for spacing and fonts in order to support responsive UI and I think the date field is too close to the patient field, given the gap between `Date:` and the input: it's easy to confuse the label & input association

File: client/packages/invoices/src/Prescriptions/DetailView/Toolbar.tsx [hunk 2]
```diff
@@ -39,7 +39,13 @@ export const Toolbar: FC = () => {
         alignItems="flex-end"
       >
         <Grid item display="flex" flex={1}>
-          <Box display="flex" flex={1} flexDirection="column" gap={1}>
+          <Box
+            display="flex"
+            flex={1}
+            flexDirection="column"
+            gap={1}
+            maxWidth={'fit-content'}
```
Comment (mark-prins) [inline client/packages/invoices/src/Prescriptions/DetailView/Toolbar.tsx:47]: nice!

**Review by mark-prins — APPROVED** [review]
looks great! thanks James 🙏 
a small tweak request, as commented.
and a query : for existing prescriptions, I understood from Andrei's comment on the issue that the created date would be used to populate the date input. I'm not seeing it in this PR though. Is it something that might happen?

**Comment by jmbrunskill** [comment]
> looks great! thanks James 🙏 a small tweak request, as commented. and a query : for existing prescriptions, I understood from Andrei's comment on the issue that the created date would be used to populate the date input. I'm not seeing it in this PR though. Is it something that might happen?

I'm just assigning this at the point where the prescription is created.
So there is an edge case where any prescriptions created before the update don't get a prescription date assigned.
I think this is ok, so will merge...

@mark-prins if you're not happy, can re-open...

**Comment by mark-prins** [comment]
ah right. yes - that's what I was testing for : editing existing prescriptions, which have an empty date field.  I was expecting that the date would be populated with the picked date of the prescription, from my reading of the comment on the issue : "And the reason to also use picked date, is because otherwise todays date would be shown on historic prescriptions ?"

**Pull request merged**